qiskit/algorithms/linear_solvers/hhl.py

```diff
@@ -0,0 +1,357 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""The HHL algorithm."""
+
+from typing import Optional, Union, List, Callable, Tuple
+import numpy as np
+
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit import QuantumCircuit, QuantumRegister, AncillaRegister
+from qiskit.circuit.library import PhaseEstimation
+from qiskit.circuit.library.arithmetic.piecewise_chebyshev import PiecewiseChebyshev
+from qiskit.opflow import Z, I, StateFn, TensoredOp
+from .exact_inverse import ExactInverse
+from .linear_solver import LinearSolver, LinearSolverResult
+from .observables.linear_system_observable import LinearSystemObservable
+
+
+# logger = logging.getLogger(__name__)
```

Comment:
I guess there are no logging statements. It can be helpful to aid debugging to have them especially if an end user has a problem you cannot reproduce then they can at least provide a log. which if its logging key information may help in resolving the problem.